mallow_notifications/sns/notification.py

```diff
@@ -0,0 +1,171 @@
+        """Sends a message using the given client and message parameters and
+        returns the response."""
+        response = client.publish(kwargs)
+        if "MessageId" in response:
+            logging.info("Notification sent successfully")
+            return response
+
+
+class SendSMS(SNS):
+    """Class for sending SMS messages using the Amazon SNS service."""
+
+    def __init__(self):
+        super().__init__()
+
+    def _check_number_status(self, phone_number_list: list, phone_number: str):
+        """Check the status of a given phone number in the phone_number_list.
+
+        :param phone_number_list: A list of phone numbers.
+        :param phone_number: The phone number to check.
+        :return: The status of the phone number, or None if not found.
+        """
+        for number in phone_number_list:
+            if number.get("PhoneNumber", None) == phone_number:
+                return number.get("Status", None)
+        return None
+
+    def send(self, kwargs):
+        """Send a message using the specified parameters.
+
+        :param kwargs: A dictionary of parameters for the message.
+        :raises NotificationError: If an error occurs while sending the
+            message.
+        """
+        try:
+            sms_sandbox_status = self.sms.get_sms_sandbox_account_status()
+            sms_sandbox_status = sms_sandbox_status.get("IsInSandbox", False)
+            if sms_sandbox_status:
+                verified_phone_numbers = self.sms.list_sms_sandbox_phone_numbers({})
+                number_status = self._check_number_status(
+                    verified_phone_numbers.get("PhoneNumbers", []),
+                    kwargs.get("phone_number", None),
+                )
+
+                if number_status == "Pending":
+                    logging.debug(
+                        "Verify your mobile number before sending the sms when you are in sandbox mode"  # pylint: disable=line-too-long
+                    )
+                    raise NotificationError(
+                        "Verify your mobile number before sending the sms when you are in sandbox mode"  # pylint: disable=line-too-long
+                    )
+
+            self._send_message(self.publish, kwargs)
+        except Exception as e:
+            raise NotificationError(f"Someting Went Wrong: {str(e)}")
+
+
+class SendTopicMessage(SNS):
+    """Class for sending topic messages using the Amazon SNS service."""
+
+    def __init__(self):
+        super().__init__()
+
+    def send(self, kwargs):
+        """Send a message using the specified parameters.
+
+        :param kwargs: A dictionary of parameters for the message.
+        :raises NotificationError: If an error occurs while sending the
+            message.
+        """
+        try:
+            self._send_message(self.publish, kwargs)
+        except Exception as e:
+            raise NotificationError(f"Someting Went Wrong: {str(e)}")
+
+
+class SendPushNotification(SNS):
```

Comment:
why some methods are private and other not ?

Reply:
In class I had helpers functions as protected method